### PR TITLE
[INLONG-6735][Sort][Manager] PostgreSQL connector supports transferring all tables for all schemas in one database

### LINKED ISSUE #6735: [Feature][Sort][Manager] PostgreSQL connector supports  transferring all tables for all schemas in one database
### Description

Inspired by mysql-cdc-inlong, postgres-cdc-inlong can also support transferring all tables in one database on time by debezium's ability.

Based on debezium-connector-postgres, postgres-cdc-inlong can assemble two data formats, which are canal-json and debezium-json. 

Postgres Logical decoding does not support DDL changes, this means that the debezium connector is unable to report DDL change events back to consumers. But postgres-cdc-inlong can do this, using JDBC query for table schema when data field changes in debezium data streams.

 Postgres-cdc-inlong also supports metrics for data transferring.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by liaorui**
### Prepare a Pull Request

- [INLONG-6735][Sort][Manager] PostgresSQL connector supports transferring all tables for all schemas in one database

- Fixes #6735

### Motivation

Inspired by mysql-cdc-inlong, postgres-cdc-inlong can also support transferring all tables in one database on time by debezium's ability.

Based on debezium-connector-postgres, postgres-cdc-inlong can assemble two data formats, which are canal-json and debezium-json.

Postgres Logical decoding does not support DDL changes, this means that the debezium connector is unable to report DDL change events back to consumers. But postgres-cdc-inlong can do this, using JDBC query for table schema when data field changes in debezium data streams.

Postgres-cdc-inlong also supports metrics for data transferring.

### Modifications

1. PostgresSQLSource class has two new parameters,  serverTimeZone and scanStartupMode. The former is Postgres Server time zone such as Aisa/Shanghai etc. The later is passed to debezium.scan.mode, the value is initial or never.

2. PostgresExtractNode also has serverTimeZone and scanStartupMode. It overwrites the getMetadataKey method to support canal-json and debezium-json meta field.

3. DebeziumChangeFetcher overwrites getTableChange method by using JDBC query for table, as debezium-connector-postgres cannot capture table changes.

4. RowDataDebeziumDeserializeSchema provides the parsers for reading tables.

5. PostgreSQLReadableMetaData provides the parsers for canal-json, debezium-json and op_type meta fields.

6. The classes in connection and manager packages are the JDBC tools to query postgres table meta info.

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Review by EMsnap — COMMENTED** [review]
Add license for copied code such as [RowDataDebeziumDeserializeSchema.java]

**Review by EMsnap — COMMENTED** [review]
[DebeziumDeserializationSchema.java] and classes in table module can be reused in mysql-cdc,  please reuse

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/FieldType.java [hunk 1]
```diff
@@ -31,6 +31,7 @@ public enum FieldType {
     FLOAT,
     DECIMAL,
     STRING,
+    TEXT,
```
Comment (yunqingmoswu) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/FieldType.java:34]: I think that there is no need to define a text here, there are already general types such as string/varchar that can be expressed.

File: inlong-sort/sort-connectors/postgres-cdc/src/main/java/org.apache.inlong.sort.cdc.postgres/debezium/internal/TableEditorImpl.java [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+import io.debezium.relational.TableId;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+public class TableEditorImpl implements TableEditor {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/postgres-cdc/src/main/java/org.apache.inlong.sort.cdc.postgres/debezium/internal/TableEditorImpl.java:33]: Please add some java comments.

File: inlong-sort/sort-connectors/postgres-cdc/src/main/java/org.apache.inlong.sort.cdc.postgres/debezium/internal/ColumnImpl.java [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+import io.debezium.relational.Column;
+import io.debezium.relational.ColumnEditor;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import io.debezium.util.Strings;
+
+public final class ColumnImpl implements Column, Comparable<Column> {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/postgres-cdc/src/main/java/org.apache.inlong.sort.cdc.postgres/debezium/internal/ColumnImpl.java:30]: Please add some java comments.

File: inlong-sort/sort-connectors/postgres-cdc/src/main/java/org.apache.inlong.sort.cdc.postgres/debezium/internal/TableImpl.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import io.debezium.annotation.PackagePrivate;
+import io.debezium.util.Strings;
+
+public final class TableImpl implements Table {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/postgres-cdc/src/main/java/org.apache.inlong.sort.cdc.postgres/debezium/internal/TableImpl.java:34]: Please add some java comments.

File: inlong-sort/sort-connectors/postgres-cdc/src/main/java/org.apache.inlong.sort.cdc.postgres/manager/PostgreSQLQueryVisitor.java [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.inlong.sort.cdc.postgres.connection.PostgreSQLJdbcConnectionProvider;
+import org.apache.inlong.sort.cdc.postgres.table.PostgreSQLDataType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PostgreSQLQueryVisitor implements Serializable {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/postgres-cdc/src/main/java/org.apache.inlong.sort.cdc.postgres/manager/PostgreSQLQueryVisitor.java:36]: Please add some java comments.

File: inlong-sort/sort-connectors/postgres-cdc/src/main/java/org.apache.inlong.sort.cdc.postgres/manager/PostgreSQLQueryVisitor.java [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+    private static final Logger LOG = LoggerFactory.getLogger(PostgreSQLQueryVisitor.class);
+
+    private final PostgreSQLJdbcConnectionProvider jdbcConnProvider;
+
+    public PostgreSQLQueryVisitor(PostgreSQLJdbcConnectionProvider jdbcConnProvider) {
+        this.jdbcConnProvider = jdbcConnProvider;
+    }
+
+    public List<Map<String, Object>> getTableColumnsMetaData(String schema, String table) {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/postgres-cdc/src/main/java/org.apache.inlong.sort.cdc.postgres/manager/PostgreSQLQueryVisitor.java:48]: Please add some comments.

File: inlong-sort/sort-connectors/postgres-cdc/src/main/java/org.apache.inlong.sort.cdc.postgres/manager/PostgreSQLQueryVisitor.java [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+            return executeQuery(query, schema, table);
+        } catch (ClassNotFoundException se) {
+            throw new IllegalArgumentException("Failed to find jdbc driver." + se.getMessage(), se);
+        } catch (SQLException se) {
+            throw new IllegalArgumentException("Failed to get table schema info from StarRocks. " + se.getMessage(),
+                    se);
+        }
+    }
+
+    public Map<String, PostgreSQLDataType> getFieldMapping(String schema, String table) {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/postgres-cdc/src/main/java/org.apache.inlong.sort.cdc.postgres/manager/PostgreSQLQueryVisitor.java:79]: Please add some comments.

File: inlong-sort/sort-connectors/postgres-cdc/src/main/java/org.apache.inlong.sort.cdc.postgres/manager/PostgreSQLQueryVisitor.java [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+        Map<String, PostgreSQLDataType> mapping = new LinkedHashMap<>();
+        for (Map<String, Object> column : columns) {
+            mapping.put(column.get("COLUMN_NAME").toString(),
+                    PostgreSQLDataType.fromString(column.get("DATA_TYPE").toString()));
+        }
+
+        return mapping;
+    }
+
+    public String getPostgreSQLVersion() {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/postgres-cdc/src/main/java/org.apache.inlong.sort.cdc.postgres/manager/PostgreSQLQueryVisitor.java:91]: Please add some java comments.

File: inlong-sort/sort-connectors/postgres-cdc/src/main/java/org.apache.inlong.sort.cdc.postgres/manager/PostgreSQLQueryVisitor.java [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+            }
+            list.add(row);
+        }
+        rs.absolute(currRowIndex);
+        rs.close();
+        jdbcConnProvider.close();
+        return list;
+    }
+
+    public Long getQueryCount(String sql) {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/postgres-cdc/src/main/java/org.apache.inlong.sort.cdc.postgres/manager/PostgreSQLQueryVisitor.java:139]: Please add some comments.

File: inlong-sort/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/cdc/postgres/table/PostgreSQLDataType.java [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+    static {
+        PostgreSQLDataType[] postgreSQLDataTypes = PostgreSQLDataType.values();
+
+        for (PostgreSQLDataType postgreSQLDataType : postgreSQLDataTypes) {
+            dataTypeMap.put(postgreSQLDataType.name(), postgreSQLDataType);
+            dataTypeMap.put(postgreSQLDataType.name().toLowerCase(), postgreSQLDataType);
+        }
+    }
+
+    public static PostgreSQLDataType fromString(String typeString) {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/cdc/postgres/table/PostgreSQLDataType.java:55]: Please add some comments.

File: inlong-sort/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/cdc/postgres/table/PostgreSQLReadableMetaData.java [hunk 11]
```diff
@@ -0,0 +1,486 @@
...
+
+        @Override
+        public Object read(SourceRecord record) {
+            Struct messageStruct = (Struct) record.value();
+            Struct sourceStruct = messageStruct.getStruct(FieldName.SOURCE);
+            return TimestampData.fromEpochMillis((Long) sourceStruct.get(AbstractSourceInfo.TIMESTAMP_KEY));
+        }
+    }),
+
+    DATA("meta.data", DataTypes.STRING(), new MetadataConverter() {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/cdc/postgres/table/PostgreSQLReadableMetaData.java:107]: Please add some comments.

File: inlong-sort/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/cdc/postgres/table/PostgreSQLReadableMetaData.java [hunk 12]
```diff
@@ -0,0 +1,486 @@
...
+            return null;
+        }
+
+        @Override
+        public Object read(SourceRecord record, @Nullable TableChange tableSchema, RowData rowData) {
+            return getCanalData(record, tableSchema, (GenericRowData) rowData);
+        }
+    }),
+
+    DATA_CANAL("meta.data_canal", DataTypes.STRING(), new MetadataConverter() {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/cdc/postgres/table/PostgreSQLReadableMetaData.java:122]: Please add some comments.

File: inlong-sort/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/cdc/postgres/table/PostgreSQLReadableMetaData.java [hunk 13]
```diff
@@ -0,0 +1,486 @@
...
+            return null;
+        }
+
+        @Override
+        public Object read(SourceRecord record, @Nullable TableChange tableSchema, RowData rowData) {
+            return getCanalData(record, tableSchema, (GenericRowData) rowData);
+        }
+    }),
+
+    DATA_DEBEZIUM("meta.data_debezium", DataTypes.STRING(), new MetadataConverter() {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/cdc/postgres/table/PostgreSQLReadableMetaData.java:137]: Please add some comments.

File: inlong-sort/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/cdc/postgres/table/PostgreSQLReadableMetaData.java [hunk 14]
```diff
@@ -0,0 +1,486 @@
...
+                public Object read(SourceRecord record) {
+                    final Envelope.Operation op = Envelope.operationFor(record);
+                    if (op != Envelope.Operation.UPDATE) {
+                        return null;
+                    }
+                    return record;
+                }
+            }),
+
+    MYSQL_TYPE("meta.mysql_type",
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/cdc/postgres/table/PostgreSQLReadableMetaData.java:286]: Please add some comments.

File: inlong-sort/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/cdc/postgres/table/PostgreSQLTableFactory.java [hunk 15]
```diff
@@ -107,6 +108,30 @@ public class PostgreSQLTableFactory implements DynamicTableSourceFactory {
                                     + "to stream events to the connector that you are configuring. Default is "
                                     + "\"flink\".");
 
+    public static final ConfigOption<String> SERVER_TIME_ZONE =
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/cdc/postgres/table/PostgreSQLTableFactory.java:111]: Please add some comments.

File: inlong-sort/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/cdc/postgres/table/PostgreSQLTableFactory.java [hunk 16]
```diff
@@ -107,6 +108,30 @@ public class PostgreSQLTableFactory implements DynamicTableSourceFactory {
...
+            ConfigOptions.key("server-time-zone")
+                    .stringType()
+                    .defaultValue("UTC")
+                    .withDescription("The session time zone in database server.");
+
+    public static final ConfigOption<String> ROW_KINDS_FILTERED =
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/cdc/postgres/table/PostgreSQLTableFactory.java:117]: Please add some comments.

File: inlong-sort/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/cdc/postgres/table/PostgreSQLTableFactory.java [hunk 17]
```diff
@@ -107,6 +108,30 @@ public class PostgreSQLTableFactory implements DynamicTableSourceFactory {
...
+                    .defaultValue("+I&-U&+U&-D")
+                    .withDescription("row kinds to be filtered,"
+                            + " here filtered means keep the data of certain row kind"
+                            + "the format follows rowKind1&rowKind2, supported row kinds are "
+                            + "\"+I\" represents INSERT.\n"
+                            + "\"-U\" represents UPDATE_BEFORE.\n"
+                            + "\"+U\" represents UPDATE_AFTER.\n"
+                            + "\"-D\" represents DELETE.");
+
+    public static final ConfigOption<Boolean> APPEND_MODE =
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/postgres-cdc/src/main/java/org/apache/inlong/sort/cdc/postgres/table/PostgreSQLTableFactory.java:129]: Please add some comments.

File: inlong-sort/sort-core/src/test/java/org/apache/inlong/sort/parser/AllMigratePostgreSQLTest.java [hunk 18]
```diff
@@ -0,0 +1,122 @@
...
+import org.apache.inlong.sort.protocol.node.format.CanalJsonFormat;
+import org.apache.inlong.sort.protocol.node.format.CsvFormat;
+import org.apache.inlong.sort.protocol.node.load.DorisLoadNode;
+import org.apache.inlong.sort.protocol.node.load.KafkaLoadNode;
+import org.apache.inlong.sort.protocol.transformation.FieldRelation;
+import org.apache.inlong.sort.protocol.transformation.relation.NodeRelation;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class AllMigratePostgreSQLTest {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-core/src/test/java/org/apache/inlong/sort/parser/AllMigratePostgreSQLTest.java:49]: Please add some comments.

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/FieldType.java (hunk 1 above)
Comment (liaorui) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/FieldType.java:34]: Postgres source connector may meet TEXT field type.

**Review by gong — APPROVED** [review]

**Comment by liaorui** [comment]
> Add license for copied code such as [RowDataDebeziumDeserializeSchema.java]

done

**Comment by liaorui** [comment]
> [DebeziumDeserializationSchema.java] and classes in table module can be reused in mysql-cdc, please reuse

done

File: inlong-sort/sort-connectors/base/pom.xml [hunk 19]
```diff
@@ -47,6 +47,13 @@
             <groupId>com.amazonaws</groupId>
             <artifactId>aws-java-sdk-s3</artifactId>
         </dependency>
+
+        <dependency>
```
[the remark this answers is not in the document]
Comment (liaorui) [reply inlong-sort/sort-connectors/base/pom.xml:51]: After removing io.debezium:debezium-embedded , sort-connector-base compiles will give many errors.

**Review by healchow — APPROVED** [review]

**Review by yunqingmoswu — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Pull request merged**